src/com/palmergames/bukkit/towny/command/PlotCommand.java

```diff
@@ -515,6 +550,163 @@ public boolean parsePlotCommand(Player player, String[] split) throws TownyExcep
 					}
 
+				} else if (split[0].equalsIgnoreCase("group")) {
+
+					TownBlock townBlock = new WorldCoord(world, Coord.parseCoord(player)).getTownBlock();
+					Town town = townBlock.getTown();
+
+					// Test we are allowed to work on this plot
+					plotTestOwner(resident, townBlock);
+					
+					if (split.length <= 1) {
+						/*
+						player.sendMessage(ChatTools.formatTitle("/... group"));
+						player.sendMessage(ChatTools.formatCommand("", "group", "add", "Ex: Groupname - \"Expensive\", \"Open\", etc..."));
+						player.sendMessage(ChatTools.formatCommand("", "group", "remove", "Ex: Groupname - \"Expensive\", \"Open\", etc..."));
+						 */
+						
+						if (townBlock.hasPlotGroup())
+							TownyMessaging.sendMessage(player, townBlock.getPlotGroup().toString());
+					}
+					
+					if (split.length >= 2) {
+						
+						if (split[1].equalsIgnoreCase("add")) {
+							
+							/*
+							PlotGroup newgrp = new PlotGroup(town.getName());
+							newgrp.setPrice(1001);
+							newgrp.setGroupName("groupname");
+							newgrp.setID(1);
+							
+							TownyWorld gWorld = TownyUniverse.getInstance().getWorldMap().get("world");
+							
+							gWorld.newGroup("test", "groupname", 420);
+							
+							townyUniverse.getDataSource().saveGroupList();
+							
+							townyUniverse.getDataSource().saveGroup(newgrp);
+							
+							 */
+							
+							TownyWorld groupWorld = townBlock.getWorld();
+							// Add the group to the new plot.
+							PlotGroup newGroup = null;
+							
+							if (townBlock.hasPlotGroup()) {
+								TownyMessaging.sendErrorMsg(player, "This plot already belongs to a group: " + townBlock.getPlotGroup().getGroupName() + ", please remove it before adding it to another.");
+								return false;
+							}
+							
+							if (split.length == 2) {
+								// The player wants to add to group using their stored mode.
+								
+								// Get the plot group from the resident mode.
+								newGroup = resident.getPlotGroupFromMode();
+								
+								if (newGroup == null) {
+									// In this case we default the group name to the player's name.
+									newGroup = new PlotGroup(town.generatePlotGroupID(), player.getName(), town);
+									
+									TownyMessaging.sendErrorMsg("Group mode was null adding new group.");
+									
+									// Add this to the town.
+									newGroup.setTown(town);
+								}
+								
+								// Check if a plot price is available.
+								if (!(townBlock.getPlotPrice() < 0)) {
+									newGroup.addPlotPrice(townBlock.getPlotPrice());
+								}
+								
+								// Set the plot group.
+								townBlock.setPlotGroup(newGroup);
+
+								// Save changes.
+								townyUniverse.getDataSource().saveTown(town);
+
+								TownyMessaging.sendMessage(player, "" + ChatColor.BLUE + townBlock.getX() + "," + townBlock.getZ() + ") was put into group " + newGroup.getGroupName());
+								
+								return true;

```

Comment:
This is block is debatable need to decide to use player's usernames as defaults for group names if they will become the owner of the plot anyways after purchase.

Reply:
To be refactored to require group names